File: app/src/main/java/com/lzm/lightLive/http/BaseResult.java
package com.lzm.lightLive.http;

/*请求返回数据 基础封装*/
public class BaseResult<T> {

    public static final int RESULT_SUCCESS = 0;

    private int errorCode;

    private String errorMsg;

    private T data;

    public BaseResult(T data) {
        this.data = data;
    }

    public BaseResult(int code, String msg) {
        this.errorCode = code;
        this.errorMsg = msg;
    }

    public int getErrorCode() {
        return errorCode;
    }

    public void setErrorCode(int errorCode) {
        this.errorCode = errorCode;
    }

    public String getErrorMsg() {
        return errorMsg;
    }

    public void setErrorMsg(String errorMsg) {
        this.errorMsg = errorMsg;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    public boolean isSuccess() {
        return RESULT_SUCCESS == errorCode;
    }

    @Override
    public String toString() {
        return "BaseResult{" +
                "errorCode=" + errorCode +
                ", errorMsg='" + errorMsg + '\'' +
                ", data=" + data +
                '}';
    }

}

File: app/src/main/java/com/lzm/lightLive/http/BaseResult.kt
package com.lzm.lightLive.http

/*请求返回数据 基础封装*/
class BaseResult<T> {
    var errorCode = 0
    var errorMsg: String? = null
    var data: T? = null
        private set

    constructor(data: T) {
        this.data = data
    }

    constructor(code: Int, msg: String?) {
        errorCode = code
        errorMsg = msg
    }

    fun setData(data: T) {
        this.data = data
    }

    val isSuccess: Boolean
        get() = RESULT_SUCCESS == errorCode

    override fun toString(): String {
        return "BaseResult{" +
                "errorCode=" + errorCode +
                ", errorMsg='" + errorMsg + '\'' +
                ", data=" + data +
                '}'
    }

    companion object {
        const val RESULT_SUCCESS = 0
    }
}